staging/src/k8s.io/cloud-provider/controllers/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -242,6 +242,7 @@ func ensureNodeExistsByProviderID(ctx context.Context, cloud cloudprovider.Inter
 			klog.Warningf("Cannot find valid providerID for node name %q, assuming non existence", node.Name)
 			return false, nil
 		}
+		node.Spec.ProviderID = providerID
```

Comment:
Alternatively we can fetch the ProviderID from the cloud provider again, like already implemented in the func [`ensureNodeExistsByProviderID`](https://github.com/kubernetes/kubernetes/blob/f3996039ef13ef75b0c91a7329f5098851e5ec34/staging/src/k8s.io/cloud-provider/controllers/nodelifecycle/node_lifecycle_controller.go#L231-L245). But to reduce API calls against the provider we should simply adjust the pointer as proposed in this PR.